### PR TITLE
chore(nimbus): apply sourcery again

### PR DESCRIPTION
**Pull request by jaredlockhart**
Because

* We still have a demo license to sourcery
* It finds common patterns for code simplification

This commit

* Applies sourcery optimizations to the backend of experimenter

### CODE REVIEW

**Review by jrbenny35 — APPROVED** [review]
Oh I love these

File: experimenter/experimenter/experiments/api/v5/serializers.py [hunk 1]
```diff
@@ -1398,20 +1391,19 @@ def _validate_schema(self, obj, schema):
...
-
+        errors = {
+            "feature_configs": [
+                f"Feature Config application {feature_config.application} "
+                f"does not match experiment application "
+                f"{self.instance.application}."
+            ]
+            for feature_config in feature_configs
+            if self.instance.application != feature_config.application
+        }
```
Comment (jaredlockhart) [inline experimenter/experimenter/experiments/api/v5/serializers.py:1406]: Wow this one is neat!  It folded three different expressions together.

**Review by mikewilli — APPROVED** [review]

File: experimenter/experimenter/experiments/api/v5/serializers.py [hunk 2]
```diff
@@ -1064,18 +1064,15 @@ def update(self, experiment, validated_data):
...
-            if (
-                not experiment.is_paused
-                and experiment.status == NimbusExperiment.Status.LIVE
-                and experiment.status_next is None
-                and experiment.publish_status == NimbusExperiment.PublishStatus.IDLE
-                and validated_data.get("publish_status")
-                != NimbusConstants.PublishStatus.REVIEW
-            ):
-                # can be Live Update (Dirty), End Enrollment, or End Experiment
-                # (including rejections) if we don't check validated_data
```
[the remark this answers is not in the document]
Comment (jaredlockhart) [reply experimenter/experimenter/experiments/api/v5/serializers.py:1077]: Yeah I noticed it nuked that.  I'll put it back.

File: experimenter/tests/integration/nimbus/utils/helpers.py [hunk 3]
```diff
@@ -160,7 +160,9 @@ def load_experiment_data(slug):
     )
 
 
-def create_basic_experiment(name, app, targeting, languages=[]):
+def create_basic_experiment(name, app, targeting, languages=None):
+    if languages is None:
+        languages = []
```
Comment (jaredlockhart) [inline experimenter/tests/integration/nimbus/utils/helpers.py:165]: Oh this is a really cool one, this is the problem in Python where you put a collection in a default arg of a function and it accidentally becomes a singleton which bleeds state between invocations @jrbenny35 That's smart.

File: cirrus/server/cirrus/experiment_recipes.py [hunk 4]
```diff
@@ -27,14 +27,16 @@ def get_recipes(self) -> dict[str, list[Any]]:
...
+        return next(
+            (
+                RecipeType.ROLLOUT.value
+                if experiment.get("isRollout", False)
+                else RecipeType.EXPERIMENT.value
+                for experiment in self.get_recipes()["data"]
+                if experiment["slug"] == experiment_slug
+            ),
+            RecipeType.EMPTY.value,
+        )
```
Comment (jaredlockhart) [inline cirrus/server/cirrus/experiment_recipes.py:39]: Whoa check that out.

Comment (jaredlockhart) [reply cirrus/server/cirrus/experiment_recipes.py]: Actually nah I don't like this I'm reverting this.  This is way less readable.

**Pull request merged**